### DISCUSSION TITLE
A correct way to use "rsre" ("remove-shadowroot-elem")

### DISCUSSION THREAD

**Discussion by DRSAgile** [uBlockO/uBO-Scriptlets#15] [Q&A] [answered]
Hi,

First of all, thanks for the scriplets.

As per instructions, my settings are:
`userResourcesLocation https://raw.githubusercontent.com/uBlock-user/uBO-Scriptlets/master/scriptlets.js`

And my use case is:
`zen.yandex.ru##+js(rsre, 'div')`
or even:
`zen.yandex.ru##+js(rsre, '*')`

The goal is to remove all shadow Divs or everything else ('*') like the blocks titled "market.yandex.ru" but it does not seem to do it on the [site](https://zen.yandex.ru) -- at least when I browse it down (though this should not matter since MutationsObserver in the script never disconnects).

But maybe I am doing something wrong?

Thanks in advance.

**Comment by uBlock-user** [comment]
Are you on Firefox ?

**Comment by DRSAgile** [comment]
Yes; version 96.0b5.

**Comment by uBlock-user** [comment, marked as answer]
https://bugzilla.mozilla.org/show_bug.cgi?id=1267027

**Comment by DRSAgile** [comment]
Thanks. So it means advertisements are unkillable in FF then. This is not the first time I stumble upon some annoying CSP limitations in this browser.

**Reply by uBlock-user** [reply]
Susbcribe to [Counters list](https://github.com/easylist/ruadlist/wiki/RU-Adlist-links) and in addition to that, on Firefox, you need to install [Scriptlet Doctor](https://github.com/JustOff/scriptlet-doctor/releases).

**Reply by DRSAgile** [reply]
I have installed everything except for "RU AdList JS Fixes" which will not work in ViolentMonkey for Firefox so it did not help much. But thank you anyway.

**Reply by uBlock-user** [reply]
You will need to take help of dimisa, maintainer of the RU AdList.
